src/envoy/utils/utils.cc

```diff
@@ -90,20 +90,27 @@ bool GetDestinationUID(const envoy::api::v2::core::Metadata& metadata,
   return true;
 }
 
-bool GetSourceUser(const Network::Connection* connection, std::string* user) {
+bool GetPrincipal(const Network::Connection* connection, bool peer,
+                  std::string* principal) {
```

Comment:
This is super confusing.. I would much rather have code duplication but have legible code in this section. You are fetching the source SAN or destination SAN. Why not make this a generic helper function and have to public functions GetSourcePrincipal and GetDestinationPrincipal ?